renderdoc/driver/metal/metal_stringise.cpp

```diff
@@ -0,0 +1,471 @@
+    MTL_STRINGISE_ENUM(TextureType1DArray);
+    MTL_STRINGISE_ENUM(TextureType2D);
+    MTL_STRINGISE_ENUM(TextureType2DArray);
+    MTL_STRINGISE_ENUM(TextureType2DMultisample);
+    MTL_STRINGISE_ENUM(TextureTypeCube);
+    MTL_STRINGISE_ENUM(TextureTypeCubeArray);
+    MTL_STRINGISE_ENUM(TextureType3D);
+    MTL_STRINGISE_ENUM(TextureType2DMultisampleArray);
+    MTL_STRINGISE_ENUM(TextureTypeTextureBuffer);
+  }
+  END_ENUM_STRINGISE()
+}
+
+template <>
+rdcstr DoStringise(const MTL::TextureUsage &el)
+{
+  BEGIN_BITFIELD_STRINGISE(MTL::TextureUsage)
+  {
+    MTL_STRINGISE_BITFIELD_BIT(TextureUsageUnknown);
+    MTL_STRINGISE_BITFIELD_BIT(TextureUsageShaderRead);
+    MTL_STRINGISE_BITFIELD_BIT(TextureUsageShaderWrite);
+    MTL_STRINGISE_BITFIELD_BIT(TextureUsageRenderTarget);
+    MTL_STRINGISE_BITFIELD_BIT(TextureUsagePixelFormatView);
+  }
+  END_BITFIELD_STRINGISE()
+}
+
+template <>
+rdcstr DoStringise(const MTL::TextureSwizzle &el)
+{
+  BEGIN_ENUM_STRINGISE(MTL::TextureSwizzle)
+  {
+    MTL_STRINGISE_ENUM(TextureSwizzleZero);
+    MTL_STRINGISE_ENUM(TextureSwizzleOne);
+    MTL_STRINGISE_ENUM(TextureSwizzleRed);
+    MTL_STRINGISE_ENUM(TextureSwizzleGreen);
+    MTL_STRINGISE_ENUM(TextureSwizzleBlue);
+    MTL_STRINGISE_ENUM(TextureSwizzleAlpha);
+  }
+  END_ENUM_STRINGISE()
+}
+
+template <>
+rdcstr DoStringise(const MTL::BlendFactor &el)
+{
+  BEGIN_ENUM_STRINGISE(MTL::BlendFactor)
+  {
+    MTL_STRINGISE_ENUM(BlendFactorZero);
+    MTL_STRINGISE_ENUM(BlendFactorOne);
+    MTL_STRINGISE_ENUM(BlendFactorSourceColor);
+    MTL_STRINGISE_ENUM(BlendFactorOneMinusSourceColor);
+    MTL_STRINGISE_ENUM(BlendFactorSourceAlpha);
+    MTL_STRINGISE_ENUM(BlendFactorOneMinusSourceAlpha);
+    MTL_STRINGISE_ENUM(BlendFactorDestinationColor);
+    MTL_STRINGISE_ENUM(BlendFactorOneMinusDestinationColor);
+    MTL_STRINGISE_ENUM(BlendFactorOneMinusDestinationAlpha);
+    MTL_STRINGISE_ENUM(BlendFactorSourceAlphaSaturated);
+    MTL_STRINGISE_ENUM(BlendFactorBlendColor);
+    MTL_STRINGISE_ENUM(BlendFactorOneMinusBlendColor);
+    MTL_STRINGISE_ENUM(BlendFactorBlendAlpha);
+    MTL_STRINGISE_ENUM(BlendFactorOneMinusBlendAlpha);
+    MTL_STRINGISE_ENUM(BlendFactorSource1Color);
+    MTL_STRINGISE_ENUM(BlendFactorOneMinusSource1Color);
+    MTL_STRINGISE_ENUM(BlendFactorSource1Alpha);
+    MTL_STRINGISE_ENUM(BlendFactorOneMinusSource1Alpha);
+  }
+  END_ENUM_STRINGISE()
+}
+
+template <>
+rdcstr DoStringise(const MTL::BlendOperation &el)
+{
+  BEGIN_ENUM_STRINGISE(MTL::BlendOperation)
+  {
+    MTL_STRINGISE_ENUM(BlendOperationAdd);
+    MTL_STRINGISE_ENUM(BlendOperationSubtract);
+    MTL_STRINGISE_ENUM(BlendOperationReverseSubtract);
+    MTL_STRINGISE_ENUM(BlendOperationMin);
+    MTL_STRINGISE_ENUM(BlendOperationMax);
+  }
+  END_ENUM_STRINGISE()
+}
+
+template <>
+rdcstr DoStringise(const MTL::ColorWriteMask &el)
+{
+  BEGIN_BITFIELD_STRINGISE(MTL::ColorWriteMask)
+  {
+    MTL_STRINGISE_BITFIELD_BIT(ColorWriteMaskNone);
```

Comment:
This (and the `All` case) should be NAMED, not BIT because they are single values to be special-cased.